### ISSUE TITLE
Compilation failure with -Werror=strict-prototypes

### ISSUE THREAD

**Issue by hhoffstaette** [lefcha/imapfilter#256]
Clang-16 will enable strict prototype checking by default, which currently fails to compile:
```
$make CC=clang CXX=clang++ CFLAGS="-pipe -march=native -O2 -Werror=strict-prototypes -I/usr/include/luajit-2.1 -DCONFIG_SHAREDIR='\"/usr/share/imapfilter\"' -DCONFIG_SSL_CAFILE='\"/etc/ssl/cert.pem\"' -DCONFIG_SSL_CAPATH='\"/etc/ssl/certs\"' " CXXFLAGS="$(echo $CFLAGS)"
...
clang -pipe -march=native -O2 -Werror=strict-prototypes -I/usr/include/luajit-2.1 -DCONFIG_SHAREDIR='"/usr/share/imapfilter"'  -DCONFIG_SSL_CAFILE='"/etc/ssl/cert.pem"' -DCONFIG_SSL_CAPATH='"/etc/ssl/certs"'    -c -o lua.o lua.c
lua.c:29:10: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
start_lua()
         ^
          void
1 error generated.
make[1]: *** [<builtin>: lua.o] Error 1
```

**Comment by hhoffstaette** [comment]
Fixed by #257

**Comment by lefcha** [comment]
Thanks for reporting, closed by f48fb06adc23315d1e437f5416cf7acde59b7779
